### PR TITLE
Concrete::getById() must return only Concrete objects

### PR DESCRIPTION
**Pull request by blankse**
Concrete::getById() returns also Folder objects. But this is wrong because the return type is static|null

### CODE REVIEW

File: models/DataObject/AbstractObject.php [hunk 1]
```diff
@@ -516,7 +516,7 @@ public static function getTotalCount($config = [])
      */
     protected static function typeMatch(AbstractObject $object)
     {
-        return in_array(static::class, [Concrete::class, __CLASS__], true) || $object instanceof static;
+        return static::class === __CLASS__ || $object instanceof static;
```
Comment (dvesh3) [inline models/DataObject/AbstractObject.php]: This change can break implementations relying on `Concrete::getById()` to load folder objects. Maybe we just trigger deprecation here and change the behavior in Pimcore 11.

Comment (blankse) [reply models/DataObject/AbstractObject.php]:
```suggestion
        if (static::class === Concrete::class && !$object instanceof static) {
            trigger_deprecation(
                'pimcore/pimcore',
                '10.5',
                'Loading none Concrete objects with via the Concrete class will not possible in Pimcore 11'
            );

            return true;
        }

        return static::class === __CLASS__ || $object instanceof static;
```

File: doc/Development_Documentation/23_Installation_and_Upgrade/09_Upgrade_Notes/README.md [hunk 2]
```diff
@@ -1,5 +1,8 @@
 # Upgrade Notes
 
+## 10.5.10
+- [DataObject] Deprecated: Loading none Concrete objects with the Concrete class will not possible in Pimcore 11.
```
Comment (jdreesen) [inline doc/Development_Documentation/23_Installation_and_Upgrade/09_Upgrade_Notes/README.md]:
```suggestion
- [DataObject] Deprecated: Loading non-Concrete objects with the Concrete class will not be possible in Pimcore 11.
```

File: models/DataObject/AbstractObject.php [hunk 3]
```diff
@@ -516,7 +516,17 @@ public static function getTotalCount($config = [])
      */
     protected static function typeMatch(AbstractObject $object)
     {
-        return in_array(static::class, [Concrete::class, __CLASS__], true) || $object instanceof static;
+        if (static::class === Concrete::class && !$object instanceof static) {
+            trigger_deprecation(
+                'pimcore/pimcore',
+                '10.5',
+                'Loading none Concrete objects with the Concrete class will not possible in Pimcore 11'
```
Comment (jdreesen) [inline models/DataObject/AbstractObject.php]:
```suggestion
                'Loading non-Concrete objects with the Concrete class will not be possible in Pimcore 11'
```

File: models/DataObject/AbstractObject.php [hunk 4]
```diff
@@ -516,7 +516,17 @@ public static function getTotalCount($config = [])
...
+            );
+
+            return true;
+        }
+
+        return static::class === __CLASS__ || $object instanceof static;
```
Comment (jdreesen) [inline models/DataObject/AbstractObject.php]: What do you think about this? It looks a bit more consistent.

```suggestion
        return static::class === self::class || $object instanceof static;
```

**Comment by dvesh3** [comment]
@blankse thanks!

**Pull request merged**